### ISSUE TITLE
Compilation error when building with stack

### ISSUE THREAD

**Issue by jwoudenberg** [fumieval/extensible#18]
Hey there! This library looks perfect for what I want to do, only I can't seem to build it. It seems to break on the `#define` macro in the `Dictionary.hs` module

Most likely I'm doing something wrong, but I have no idea what. I'm getting a compilation error when adding this library as a dependency and buiding using stack. I've tried this using stack 1.6.1 and 1.6.3, using their lts-9.21 (GHC 8.0.2) and lts-10.2 (GHC 8.2.2), but the effect is the same across all these versions. I'm on OSX. This is the compilation log of the errors.

Any help figuring out what's going on here would be greatly appreciated!

```
--  While building custom Setup.hs for package extensible-0.4.7 using:
      /Users/jasper/.stack/setup-exe-cache/x86_64-osx/Cabal-simple_mPHDZzAJ_2.0.1.0_ghc-8.2.2 --builddir=.stack-work/dist/x86_64-osx/Cabal-2.0.1.0 build --ghc-options " -ddump-hi -ddump-to-file -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
    Logs have been written to: /Users/jasper/dev/elm-gen/.stack-work/logs/extensible-0.4.7.log

    Configuring extensible-0.4.7...
    Preprocessing library for extensible-0.4.7..
    Building library for extensible-0.4.7..
    [ 1 of 23] Compiling Data.Extensible.Internal ( src/Data/Extensible/Internal.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Internal.o )

    /private/var/folders/x3/4hf6y2j97xx_4y57ds425v_r0000gn/T/stack53745/extensible-0.4.7/src/Data/Extensible/Internal.hs:68:1: warning: [-Wunused-imports]
        The import of ‘Data.Bits’ is redundant
          except perhaps to import instances from ‘Data.Bits’
        To import instances alone, use: import Data.Bits()
       |
    68 | import Data.Bits
       | ^^^^^^^^^^^^^^^^
    [ 2 of 23] Compiling Data.Extensible.HList ( src/Data/Extensible/HList.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/HList.o )
    [ 3 of 23] Compiling Data.Extensible.Internal.Rig ( src/Data/Extensible/Internal/Rig.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Internal/Rig.o )
    [ 4 of 23] Compiling Data.Extensible.Wrapper ( src/Data/Extensible/Wrapper.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Wrapper.o )
    [ 5 of 23] Compiling Data.Extensible.Class ( src/Data/Extensible/Class.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Class.o )
    [ 6 of 23] Compiling Data.Extensible.Sum ( src/Data/Extensible/Sum.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Sum.o )
    [ 7 of 23] Compiling Data.Extensible.Struct ( src/Data/Extensible/Struct.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Struct.o )
    [ 8 of 23] Compiling Data.Extensible.Product ( src/Data/Extensible/Product.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Product.o )
    [ 9 of 23] Compiling Data.Extensible.Plain ( src/Data/Extensible/Plain.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Plain.o )
    [10 of 23] Compiling Data.Extensible.Match ( src/Data/Extensible/Match.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Match.o )
    [11 of 23] Compiling Data.Extensible.Inclusion ( src/Data/Extensible/Inclusion.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Inclusion.o )
    [12 of 23] Compiling Data.Extensible.Nullable ( src/Data/Extensible/Nullable.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Nullable.o )
    [13 of 23] Compiling Data.Extensible.Field ( src/Data/Extensible/Field.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Field.o )
    [14 of 23] Compiling Data.Extensible.Tangle ( src/Data/Extensible/Tangle.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Tangle.o )
    [15 of 23] Compiling Data.Extensible.Record ( src/Data/Extensible/Record.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Record.o )

    /private/var/folders/x3/4hf6y2j97xx_4y57ds425v_r0000gn/T/stack53745/extensible-0.4.7/src/Data/Extensible/Record.hs:63:5: warning: [-Wunused-matches]
        Defined but not used: ‘rec’
       |
    63 |     rec <- newName "rec"
       |     ^^^
    [16 of 23] Compiling Data.Extensible.Label ( src/Data/Extensible/Label.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Label.o )
    [17 of 23] Compiling Data.Extensible.GetOpt ( src/Data/Extensible/GetOpt.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/GetOpt.o )
    [18 of 23] Compiling Data.Extensible.Effect ( src/Data/Extensible/Effect.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Effect.o )
    [19 of 23] Compiling Data.Extensible.TH ( src/Data/Extensible/TH.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/TH.o )
    [20 of 23] Compiling Data.Extensible.Effect.Default ( src/Data/Extensible/Effect/Default.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Effect/Default.o )
    [21 of 23] Compiling Data.Extensible.Dictionary ( src/Data/Extensible/Dictionary.hs, .stack-work/dist/x86_64-osx/Cabal-2.0.1.0/build/Data/Extensible/Dictionary.o )

    /private/var/folders/x3/4hf6y2j97xx_4y57ds425v_r0000gn/T/stack52512/extensible-0.4.7/src/Data/Extensible/Dictionary.hs:104:98: error:
        • Variable not in scope: expr :: Membership (x : xs) x1 -> m a6
        • Perhaps you meant ‘exp’ (imported from Prelude)
        |
    104 |     = ENUM_EACH(\i -> G.basicUnsafeCopy (hlookupC i v) (hlookupC i w))
        |                                                                                                  ^^^^

    /private/var/folders/x3/4hf6y2j97xx_4y57ds425v_r0000gn/T/stack52512/extensible-0.4.7/src/Data/Extensible/Dictionary.hs:119:127: error:
        • Variable not in scope: expr :: Membership (x : xs) x1 -> m a0
        • Perhaps you meant ‘exp’ (imported from Prelude)
        |
    119 |   basicInitialize (MV_Product v) = ENUM_EACH(\i -> M.basicInitialize $ hlookupC i v)
        |                                                                                                                               ^^^^

    /private/var/folders/x3/4hf6y2j97xx_4y57ds425v_r0000gn/T/stack52512/extensible-0.4.7/src/Data/Extensible/Dictionary.hs:126:132: error:
        • Variable not in scope: expr :: Membership (x : xs) x1 -> m a1
        • Perhaps you meant ‘exp’ (imported from Prelude)
        |
    126 |   basicUnsafeWrite (MV_Product v) i x = ENUM_EACH(\m -> M.basicUnsafeWrite (hlookupC m v) i (hlookup m x))
        |                                                                                                                                    ^^^^

    /private/var/folders/x3/4hf6y2j97xx_4y57ds425v_r0000gn/T/stack52512/extensible-0.4.7/src/Data/Extensible/Dictionary.hs:127:122: error:
        • Variable not in scope: expr :: Membership (x : xs) x1 -> m a2
        • Perhaps you meant ‘exp’ (imported from Prelude)
        |
    127 |   basicClear (MV_Product v) = ENUM_EACH(\i -> M.basicClear $ hlookupC i v)
        |                                                                                                                          ^^^^

    /private/var/folders/x3/4hf6y2j97xx_4y57ds425v_r0000gn/T/stack52512/extensible-0.4.7/src/Data/Extensible/Dictionary.hs:128:122: error:
        • Variable not in scope: expr :: Membership (x : xs) x1 -> m a3
        • Perhaps you meant ‘exp’ (imported from Prelude)
        |
    128 |   basicSet (MV_Product v) x = ENUM_EACH(\i -> M.basicSet (hlookupC i v) (hlookup i x))
        |                                                                                                                          ^^^^

    /private/var/folders/x3/4hf6y2j97xx_4y57ds425v_r0000gn/T/stack52512/extensible-0.4.7/src/Data/Extensible/Dictionary.hs:130:98: error:
        • Variable not in scope: expr :: Membership (x : xs) x1 -> m a4
        • Perhaps you meant ‘exp’ (imported from Prelude)
        |
    130 |     = ENUM_EACH(\i -> M.basicUnsafeCopy (hlookupC i v1) (hlookupC i v2))
        |                                                                                                  ^^^^

    /private/var/folders/x3/4hf6y2j97xx_4y57ds425v_r0000gn/T/stack52512/extensible-0.4.7/src/Data/Extensible/Dictionary.hs:132:98: error:
        • Variable not in scope: expr :: Membership (x : xs) x1 -> m a5
        • Perhaps you meant ‘exp’ (imported from Prelude)
        |
    132 |     = ENUM_EACH(\i -> M.basicUnsafeMove (hlookupC i v1) (hlookupC i v2))
        |                                                                                                  ^^^^
```

**Comment by fumieval** [comment]
Thanks for reporting; it looks like it's caused by a recent update to OSX. I expanded those macros for now, and released v0.4.7.1. I'm still investigating what's going on.

**Comment by jwoudenberg** [comment]
Awesome, thanks for the quick response!
